src/rules/prefer-spy-on.ts

```diff
@@ -69,6 +74,19 @@ export default createRule({
 
         if (!jestFnCall) return;
 
+        // The `jest.fn()` must be in a `describe` block (so this includes all
+        // hook blocks) or `test` block (since they can be outside `describe`
+        // blocks) in order for us to report it.
+        const ancestors = context.getAncestors();
+        const isInTestBlock = ancestors.some(
+          ancestor =>
+            ancestor.type === AST_NODE_TYPES.CallExpression &&
+            (ancestor.callee.name in DescribeAlias ||
```

Comment:
You might has missed my previous comment - sorry for the double up if you did see it before pushing!

`in` will match `hasOwnProperty`, which is why we switched from `in` to use `hasOwnProperty`.

But you can actually just replace this w/ the `isTestCase` & `isDescribe` guards we have in `utils` :)